datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/TableGenerator.scala

```diff
@@ -0,0 +1,270 @@
+    - c_foreign_a Should overlap with a_facts.primary_a about 50% of the time
+    - key group 2 columns up to the complexity (each key should show up about 10 times)
+    - c_data_row_num_1 1 data column that is essentially the row number for order by tests in window
+    - 5 data columns
+    - 5 numeric data columns
+   */
+  private def genCData(): DataFrame = {
+    val schema = "c_foreign_a long," +
+      (1 to complexity).map(i => s"c_key2_$i ${randomColumnType()}").mkString(",") + "," +
+      "c_data_row_num_1 long," +
+      (1 to 5).map(i => s"c_data_$i ${randomColumnType()}").mkString(",") + "," +
+      (1 to 5).map(i => s"c_data_numeric_$i ${randomNumericColumnType()}").mkString(",")
+    val overlapRatio = 0.50
+    val cData = dbgen.addTable("c_data", schema, cNumRows)
+    val overlapSeeds = Math.round(aNumRows * overlapRatio)
+    val startSeed = aNumRows - overlapSeeds + 1
+    val endSeed = startSeed + aNumRows
+    cData("c_foreign_a").setSeedRange(startSeed, endSeed)
+    cData.configureKeyGroup(
+      (1 to complexity).map(i => s"c_key2_$i"),
+      CorrelatedKeyGroup(1, 1, 10),
+      FlatDistribution())
+    cData("c_data_row_num_1").setValueGen(RowNumPassThrough)
+    cData.toDF(spark)
+  }
+  /**
+  d_data: Each scale factor corresponds to 100,000 rows.
+    - key group 2 complexity columns (each key should show up about 10 time, but the overlap with c_data for key group 2 should only be about 50%)
+    - 10 data columns
+   */
+  private def genDData(): DataFrame = {
+    val schema = (1 to complexity).map(i => s"d_key2_$i ${randomColumnType()}").mkString(",") +
+      "," +
+      (1 to 10).map(i => s"d_data_$i ${randomColumnType()}").mkString(",")
+    val dData = dbgen.addTable("d_data", schema, dNumRows)
+    // each key should show up about 10 time, but the overlap with c_data for key group 2 should only be about 50%
+    // c_data key group uses seed range (1, 10) so here d_data uses (6, 15) to get 50% overlap
+    dData.configureKeyGroup(
+      (1 to complexity).map(i => s"d_key2_$i"),
+      CorrelatedKeyGroup(1, 6, 15),
+      FlatDistribution()
+    )
+    dData.toDF(spark)
+  }
+  /**
+  e_data: Each scale factor corresponds to 1,000,000 rows.
+    - key group 3 with 3 columns (the unique count should be about 99% of the total number of rows and overlap with b_data key group 3 by about 90%).
+    - 10 data columns
+   */
+  private def genEData(): DataFrame = {
+    val schema = "e_key3_1 string," +
+      "e_key3_2 date," +
+      "e_key3_3 long," +
+      (1 to 10).map(i => s"e_data_$i ${randomColumnType()}").mkString(",")
+    val overlapRatio = 0.9
+    val eData = dbgen.addTable("e_data", schema, eNumRows)
+    val overlapBias = Math.round(bNumRows * (1 - overlapRatio))
+    val startSeed = 1 + overlapBias
+    val endSeed = Math.round(0.99 * eNumRows) + overlapBias
+    eData.configureKeyGroup(
+      (1 to 3).map(i => s"e_key3_$i"),
+      CorrelatedKeyGroup(1, startSeed, endSeed),
```

Comment:
nit: could we have the keygroup ids match?

```
CorrelatedKeyGroup(3, startSeed, endSeed)
```

Reply:
Updated.